### PR TITLE
Refactor disable preview toggle to be only front end

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Description

Related to: https://github.com/kedro-org/kedro-viz/issues/2042

This ticket is to refactor the preview datasets toggle setting feature to handle logic for previews from FE instead of making BE calls.

This feature enables users to control data previews for all nodes from the settings panel of Kedro-Viz when running kedro viz locally.

After running Kedro-Viz locally, dataset previews for all nodes will be enabled by default. However, users will have the ability to enable or disable this feature from the Kedro-Viz settings panel.

## Development notes

We now have a check for the dataset preview toggle in `metadata.js` before rendering the previews.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

File: src/components/metadata/metadata.js [hunk 1]
```diff
@@ -266,7 +267,7 @@ const MetaData = ({
                   </>
                 )}
               </dl>
-              {hasPlot && (
+              {hasPlot && showDatasetPreviews && (
```
Comment (rashidakanchwala) [inline src/components/metadata/metadata.js:270]: maybe you add it just here 

const hasPreview = showDatasetPreviews && metadata?.preview;

**Review by rashidakanchwala — APPROVED** [review]
Just one comment otherwise looks good :)

**Comment by ravi-kumar-pilla** [comment]
Hi @SajidAlamQB , Thank you for the refactor. Tested this on Gitpod and it works well. 

I just have 1 question for the flag in the UI. Do we show this flag even on the hosted Kedro-Viz instance ? Like if someone does a `kedro viz deploy` or `kedro viz build` without including preview, will Kedro-Viz has the flag in settings panel ? 

For me the flag does make sense in local as the user cannot disable preview on `kedro viz run`. So the flag is always `True` by default. But for the deployed version,  user can control this behavior using `--include-previews` . The flag will not have any effect if someone did not include preview when building the viz instance. cc: @rashidakanchwala @stephkaiser 

Thank you

**Comment by SajidAlamQB** [comment]
> For me the flag does make sense in local as the user cannot disable preview on `kedro viz run`. So the flag is always `True` by default. But for the deployed version, user can control this behavior using `--include-previews` . The flag will not have any effect if someone did not include preview when building the viz instance. cc: @rashidakanchwala @stephkaiser

Good spot! As it is, the flag will be shown even if the `--include-previews` argument isn't used, meaning the flag might not function as expected without it. I think we can fix this by conditionally displaying the flag in the settings panel only if the `--include-previews` argument was used during deployment. What do you think? Is there any existing flags we do this for that we can take inspiration from?

**Review by jitu5 — APPROVED** [review]
Thanks @SajidAlamQB

**Comment by ravi-kumar-pilla** [comment]
> Good spot! As it is, the flag will be shown even if the `--include-previews` argument isn't used, meaning the flag might not function as expected without it. I think we can fix this by conditionally displaying the flag in the settings panel only if the `--include-previews` argument was used during deployment. What do you think? Is there any existing flags we do this for that we can take inspiration from?

One way we can do this is using the viz_metadata_file that is available in the build/deploy artifacts. 

At this moment, it has timestamp and version keys. May be adding a json key like `is_all_previews_enabled` in this file and in the FE reading the json in deployment mode (i.e., not localhost) could be an option. But I am happy to hear from the team. Thank you

**Comment by rashidakanchwala** [comment]
fair points, I would keep this simple for now the way it is. and wait for user to raise this if it's an issue and then try to fix it.

**Pull request merged**